fix: multiple mutations not calling callbacks each time

wtanna opened the pull request:
By having `isLatest` be a function it was only setting the `data` if it
was the last mutation being run, which then called all the callbacks with that 
one set of data. With this adjustment every mutation will call every callback.

This change fixes this issue: https://github.com/tannerlinsley/react-query/issues/241

tannerlinsley commented:
I fixed this in a bit of a different way, but the test here is great. Can you isolate the test and drop the rest of the changes and then we can merge this?

Commit:
Add callback tests for useMutation

Adds callback tests for when the `mutate` function is called multiple
times in a row.

tannerlinsley commented:
:tada: This PR is included in version 2.4.2 :tada:

The release is available on:
- [npm package (@latest dist-tag)](https://www.npmjs.com/package/react-query/v/2.4.2)
- [GitHub release](https://github.com/tannerlinsley/react-query/releases/tag/v2.4.2)

Your **[semantic-release](https://github.com/semantic-release/semantic-release)** bot :package::rocket:
